pkg/agent/proxy/proxier.go

```diff
@@ -1232,3 +1232,87 @@ func NewDualStackProxier(
 
 	return &metaProxierWrapper{ipv4Proxier, ipv6Proxier, metaProxier}, nil
 }
+
+func NewProxier(
+	hostname string,
+	proxyName string,
+	k8sClient clientset.Interface,
+	ofClient openflow.Client,
+	v4Enabled bool,
+	v6Enabled bool,
+	routeClient route.Interface,
+	nodePortAddressesIPv4 []net.IP,
+	nodePortAddressesIPv6 []net.IP,
+	proxyAllEnabled bool,
+	skipServices []string,
+	proxyLoadBalancerIPs bool,
+	v4GroupCounter types.GroupCounter,
+	v6GroupCounter types.GroupCounter,
+	nestedServiceSupport bool,
+	stopCh <-chan struct{}) (Proxier, error) {
+
+	optsModifier, err := getServiceListOptionsModifier(proxyName)
+	if err != nil {
+		return nil, err
+	}
+	proxyInformerFactory := informers.NewSharedInformerFactoryWithOptions(k8sClient, informerDefaultResync, informers.WithTweakListOptions(optsModifier))
+
+	var proxier Proxier
+	switch {
+	case v4Enabled && v6Enabled:
+		proxier, err = NewDualStackProxier(hostname,
```

Comment:
Does NewDualStackProxier still need to be public?

Reply:
I missed this, thanks. I have updated this to private.